intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,462 @@
+# ------------------------------------------------
+#
+# ``torch.export`` takes in a callable (including ``torch.nn.Module``s),
+# a tuple of positional arguments, and optionally (not shown in the example below),
+# a dictionary of keyword arguments.
+
+import torch
+from torch.export import export
+
+class MyModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.lin = torch.nn.Linear(100, 10)
+
+    def forward(self, x, y):
+        return torch.nn.functional.relu(self.lin(x + y), inplace=True)
+
+mod = MyModule()
+exported_mod = export(mod, (torch.randn(8, 100), torch.randn(8, 100)))
+print(type(exported_mod))
+
+######################################################################
+# ``torch.export`` returns an ``ExportedProgram``, which is not a ``torch.nn.Module``,
+# but can still be ran as a function:
+
+print(exported_mod(torch.randn(8, 100), torch.randn(8, 100)))
+
+######################################################################
+# ``ExportedProgram`` has some attributes that are of interest.
+# The ``graph`` attribute is an FX graph traced from the function we exported,
+# that is, the computation graph of all PyTorch operations.
+# The FX graph has some important properties:
+# - The operations are "ATen-level" operations.
+# - The graph is "functionalized", meaning that no operations are mutations.
+#
+# The ``graph_module`` attribute is the ``GraphModule`` that wraps the ``graph`` attribute
+# so that it can be ran as a ``torch.nn.Module``.
+# We can use ``graph_module``'s ``print_readable` to print a Python code representation
+# of ``graph``:
+
+print(exported_mod)
+exported_mod.graph_module.print_readable()
+
+######################################################################
+# The printed code shows that FX graph only contains ATen-level ops (i.e. ``torch.ops.aten``)
+# and that mutations were removed (e.g. the mutating op ``torch.nn.functional.relu(..., inplace=True)``
+# is represented in the printed code by ``torch.ops.aten.relu.default``, which does not mutate).
+
+######################################################################
+# Other attributes of interest in ``ExportedProgram`` include:
+# - ``graph_signature`` -- the inputs, outputs, parameters, buffers, etc. of the exported graph.
+# - ``range_constraints`` and ``equality_constraints`` -- Constraints, covered later
+
+print(exported_mod.graph_signature)
+
+######################################################################
+# Comparison to ``torch.compile``
+# -------------------------------
+# Although ``torch.export`` is built on top of the ``torch.compile``
+# components, the key limitation of ``torch.export`` is that it does not
+# support graph breaks. This is because handling graph breaks involves interpreting
+# the unsupported operation with default Python evaluation, which is incompatible
+# with the export use case.
+#
+# A graph break is necessary in cases such as:
+# - data-dependent control flow
```

Comment:
Are there other cases for graph breaks? Like unsupported python operations? Otherwise this doesn't need to be a list.

EDIT: looks like other examples are below, perhaps the formatting should be improved for this first section?